app/src/main/java/cl/freedom/wheatermvvm/dashboard/ui/CurrentCovidViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package cl.freedom.wheatermvvm.dashboard.ui
+
+import androidx.lifecycle.*
+import cl.freedom.desafiomvvm.util.toast
+import cl.freedom.wheatermvvm.dashboard.data.CurrentCovidMapper
+import cl.freedom.wheatermvvm.dashboard.data.CurrentCovidRepository
+import cl.freedom.wheatermvvm.data.CovidListener
+import cl.freedom.wheatermvvm.data.Result
+import cl.freedom.wheatermvvm.data.response.CovidResponse
+import cl.freedom.wheatermvvm.util.State
+import javax.inject.Inject
+
+class CurrentCovidViewModel @Inject constructor(private val repository: CurrentCovidRepository) : ViewModel()
+{
+    var covidListener : CovidListener? = null
```

Comment:
Why you are handling the listener on the viewModel and not on the UI ?

Reply:
it listener was deleted